### PR TITLE
refactor(kit): new version of `InputNumber`

### PR DESCRIPTION
**Pull request by nsbarsukov**

### CODE REVIEW

File: projects/demo/src/modules/app/app.routes.ts [hunk 1]
```diff
@@ -424,6 +424,11 @@ export const ROUTES: Routes = [
         loadComponent: async () => import('../components/input-number'),
```
Comment (nsbarsukov) [inline projects/demo/src/modules/app/app.routes.ts:424]: Hide this route before merge

File: projects/demo-playwright/tests/legacy/input-number/input-number.pw.spec.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ test.describe('InputNumber', () => {
         });
 
         test('Infinite precision', async ({page}) => {
-            await tuiGoto(page, `${DemoRoute.InputNumber}/API?precision=Infinity`);
+            await tuiGoto(page, `${DemoRoute.InputNumberLegacy}/API?precision=Infinity`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/legacy/input-number/input-number.pw.spec.ts:25]: All changes inside this file – `DemoRoute.InputNumber` => `DemoRoute.InputNumberLegacy` + tabulation. Nothing more.

File: projects/demo/src/modules/components/input-number-legacy/examples/1/index.html
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-number-legacy/examples/1/index.html:1]: No changes inside `projects/demo/src/modules/components/input-number-legacy/**`-path. Only folder renaming

File: projects/kit/components/input-number/input-number.component.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    });
+
+    private readonly min = signal(this.options.min);
+    private readonly max = signal(this.options.max);
+    private readonly prefix = signal(this.options.prefix);
+    private readonly postfix = signal(this.options.postfix);
+
+    private readonly precision = computed(() =>
+        Number.isNaN(this.numberFormat().precision) ? 2 : this.numberFormat().precision,
+    );
```
Comment (nsbarsukov) [inline projects/kit/components/input-number/input-number.component.ts:71]: Should we make BC for this logic ? 🤔 

```suggestion
    private readonly precision = computed(() =>
        Number.isNaN(this.numberFormat().precision) ? 0 : this.numberFormat().precision,
    );
```

**Review by splincode — APPROVED** [review]

File: projects/demo/src/components/control/index.ts [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+    inject,
+    Input,
+} from '@angular/core';
+import {NgControl} from '@angular/forms';
 import {TuiDocAPIItem} from '@taiga-ui/addon-doc';
 import {TuiTitle} from '@taiga-ui/core';
 
+@Directive({
+    standalone: true,
+    selector: 'input[tuiDisabled][formControl]',
```
Comment (waterplea) [inline projects/demo/src/components/control/index.ts:15]: I already have this directive here:
https://github.com/taiga-family/taiga-ui/blob/main/projects/demo/src/utils/disabled.directive.ts

File: projects/kit/components/input-number/input-number.options.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import type {Provider} from '@angular/core';
```
Comment (waterplea) [inline projects/kit/components/input-number/input-number.options.ts:1]: Let's use this util to create token and helper:
https://github.com/taiga-family/taiga-ui/blob/main/projects/cdk/utils/di/create-options.ts

File: projects/kit/components/input-number/input-number.options.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import type {TuiValueTransformer} from '@taiga-ui/cdk/classes';
+import {tuiCreateToken, tuiProvideOptions} from '@taiga-ui/cdk/utils/miscellaneous';
+
+export interface TuiInputNumberOptions {
+    readonly max: number;
+    readonly min: number;
+    readonly prefix: string;
+    readonly postfix: string;
+    readonly valueTransformer: TuiValueTransformer<number | null> | null;
```
Comment (waterplea) [inline projects/kit/components/input-number/input-number.options.ts:10]: Are we sure we're introducing it here?

File: projects/kit/components/input-number/input-number.component.ts [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+const DEFAULT_MAX_LENGTH = 18;
+
+@Component({
+    standalone: true,
+    selector: 'input[tuiInputNumber]',
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [tuiFallbackValueProvider(null)],
+    hostDirectives: [
+        // TODO: replace with `TuiWithTextfield` after merging of https://github.com/taiga-family/taiga-ui/pull/9976
```
Comment (waterplea) [inline projects/kit/components/input-number/input-number.component.ts:39]: Merged

File: projects/kit/components/input-number/input-number.component.ts [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+        const value = maskitoParseNumber(
+            this.textfieldValue(),
+            this.numberFormat().decimalSeparator,
+        );
+
+        return value < 0 ? value > this.max() : value < this.min();
+    });
+
+    protected textfieldValue = signal(this.element.value || '');
+    protected override readonly transformer = this.options.valueTransformer;
```
Comment (waterplea) [inline projects/kit/components/input-number/input-number.component.ts:83]: Maybe instead we should use DI:
```ts
{
  provide: TuiValueTransformer,
  useFactory: () => inject(TUI_INPUT_NUMBER_OPTIONS).valueTransformer`
}
```
If we plan to do this often with other components we can make a helper `tuiValueTransformerFrom(token)`

File: projects/kit/components/input-number/input-number.component.ts [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+
+    // TODO(v5): replace with signal input
+    @Input('postfix')
+    public set postfixSetter(x: string) {
+        this.postfix.set(x);
+    }
+
+    public override writeValue(value: number | null): void {
+        super.writeValue(value);
+        this.textfieldValue.set(this.formatNumber(value));
```
Comment (waterplea) [inline projects/kit/components/input-number/input-number.component.ts:142]: I wonder if there's a way to make textfieldValue a computed based on value and isIndetermediateState? Are we going to need this type of signal in other masked controls? Maybe it should be part of `TuiControl` class?

File: projects/demo/src/components/control/index.ts (hunk 4 above)
Comment (nsbarsukov) [reply projects/demo/src/components/control/index.ts:15]: Oops 😄

Comment (nsbarsukov) [reply projects/demo/src/components/control/index.ts]: Fixed https://github.com/taiga-family/taiga-ui/pull/10031/commits/21703a58cae39021aef32a9ecf4cb297188a55ec

File: projects/kit/components/input-number/input-number.options.ts (hunk 5 above)
Comment (nsbarsukov) [reply projects/kit/components/input-number/input-number.options.ts]: Fixed in https://github.com/taiga-family/taiga-ui/pull/10031/commits/21703a58cae39021aef32a9ecf4cb297188a55ec

File: projects/kit/components/input-number/input-number.component.ts (hunk 7 above)
Comment (nsbarsukov) [reply projects/kit/components/input-number/input-number.component.ts]: Fixed in https://github.com/taiga-family/taiga-ui/pull/10031/commits/21703a58cae39021aef32a9ecf4cb297188a55ec

File: projects/kit/components/input-number/input-number.component.ts (hunk 8 above)
Comment (nsbarsukov) [reply projects/kit/components/input-number/input-number.component.ts]: https://github.com/taiga-family/taiga-ui/pull/10031/commits/f6354bee6b1718e61eca569a25e86ccfd835a29d

**Review by waterplea — APPROVED** [review]

**Pull request merged**